Fix aria-label documentation, removing statement that says it should be used only for interactive items

phsantiago opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description
This change closes https://github.com/mdn/content/issues/28397, by changing the first paragraph of aria-label page, based on aria-label spec https://w3c.github.io/aria/#aria-label.
<!-- ✍️ Summarize your changes in one or two sentences -->

### Motivation
There are noninteractive elements, such as List, Img, and Meter, that allow aria-label, and it's useful to have this context announced when using assistive technologies.
- https://w3c.github.io/aria/#list aria-label is present in "Inherited States and Properties"
- https://w3c.github.io/aria/#img aria-label is present in "Inherited States and Properties"
- https://w3c.github.io/aria/#meter aria-label is present in "Inherited States and Properties"

<!-- ❓ Why are you making these changes and how do they help readers? -->

### Additional details
The confusion might have appeared because there are multiple non interactive aria roles, which aria-label is forbidden:

> Used in Roles: All elements of the base markup except for the following roles: [caption](https://w3c.github.io/aria/#caption), [code](https://w3c.github.io/aria/#code), [definition](https://w3c.github.io/aria/#definition), [deletion](https://w3c.github.io/aria/#deletion), [emphasis](https://w3c.github.io/aria/#emphasis), [generic](https://w3c.github.io/aria/#generic), [insertion](https://w3c.github.io/aria/#insertion), [mark](https://w3c.github.io/aria/#mark), [none](https://w3c.github.io/aria/#none), [paragraph](https://w3c.github.io/aria/#paragraph), [strong](https://w3c.github.io/aria/#strong), [subscript](https://w3c.github.io/aria/#subscript), [suggestion](https://w3c.github.io/aria/#suggestion), [superscript](https://w3c.github.io/aria/#superscript), [term](https://w3c.github.io/aria/#term), [time](https://w3c.github.io/aria/#time)
<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

### Related issues and pull requests
https://github.com/mdn/content/issues/28397
<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

phsantiago committed:
updates first paragraph of aria-label

scottaohara commented:
Might make sense to explicitly mention aria-label can be used to name an element, so long as the elements implicit or explicit aria role doesn’t prohibit naming.  

Listitem, for example being a role that is incorrectly not in the name prohibited category, but will be once the spec updates to correct this. But the other example, role=list or main, article, etc are all nameable roles/elements.

phsantiago committed:
states clearly that it is meant to be used with roles that allow author naming

phsantiago commented on files/en-us/web/accessibility/aria/attributes/aria-label/index.md:
```diff
@@ -7,7 +7,7 @@ spec-urls: https://w3c.github.io/aria/#aria-label
 {{AccessibilitySidebar}}
 
-The `aria-label` attribute defines a string value that labels an interactive element.
+The `aria-label` attribute defines a string value that names an element.
```
> Might it make sense to explicitly mention that aria-label can be used to name an element, so long as the element's implicit or explicit aria role doesn’t prohibit naming.

Pushed a change https://github.com/mdn/content/pull/35465/commits/83297c784da87276c534ff5019dcca5f761f6de3 making it clearer that label should only be used with roles that don't prohibit author naming.

>Listitem, for example being a role that is incorrectly not in the name prohibited category, but will be once the spec updates to correct this. But the other example, role=list or main, article, etc are all nameable roles/elements.

Thanks for the context, @scottaohara! I found this nice PR of you on the theme: https://github.com/w3c/aria/pull/2292. Let me know if I could help with anything there, too   🙂

phsantiago commented on files/en-us/web/accessibility/aria/attributes/aria-label/index.md:
```diff
@@ -7,13 +7,13 @@ spec-urls: https://w3c.github.io/aria/#aria-label
 {{AccessibilitySidebar}}
 
-The `aria-label` attribute defines a string value that labels an interactive element.
+The `aria-label` attribute defines a string value that can be used to name an element, as long as the element's role does not [prohibit naming](/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label#associated_roles).
```
> Might it make sense to explicitly mention that aria-label can be used to name an element, so long as the element's implicit or explicit aria role doesn’t prohibit naming.

Pushed a change https://github.com/mdn/content/pull/35465/commits/83297c784da87276c534ff5019dcca5f761f6de3 making it clearer that label should only be used with roles that don't prohibit author naming.

>Listitem, for example being a role that is incorrectly not in the name prohibited category, but will be once the spec updates to correct this. But the other example, role=list or main, article, etc are all nameable roles/elements.

Thanks for the context, @scottaohara! I found this nice PR of you on the theme: https://github.com/w3c/aria/pull/2292. Let me know if I could help with anything there, too   🙂

phsantiago committed:
Merge branch 'mdn:main' into patch-1

phsantiago committed:
remove pathname from same page anchor links

phsantiago commented:
@scottaohara I’ve addressed the feedback provided in the initial review of PR. Could you please take some time to re-review? Thanks in advance 🙂

scottaohara commented on files/en-us/web/accessibility/aria/attributes/aria-label/index.md:
````diff
@@ -32,7 +32,7 @@ The code below shows an example of how to use the `aria-label` attribute to prov
 ```
 
-> **Note:** `aria-label` is intended for interactive elements or for elements made interactive via other ARIA declarations when there's no visible text in the DOM to serve as a label.
+> **Note:** `aria-label` is intended for naming elements that allow `aria-label` or for elements made interactive via other ARIA declarations when there's no visible text in the DOM to serve as a label.
````
```suggestion
> **Note:** `aria-label` is intended for naming elements where the implicit or explicit role does not prohibit naming. It is strongly recommended to prioritize the use of `aria-labelledby` over `aria-label` if a visible label exists for the element to reference and receive its name from. 
```

phsantiago committed:
accept change, mentioning implicit or explicit role does not prohibit naming

Co-authored-by: Scott O'Hara <scottaohara@users.noreply.github.com>

phsantiago committed:
mdn-lint: remove trailing space

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

phsantiago commented:
Hi @scottaohara, thanks for your previous review. I applied the change you suggested https://github.com/mdn/content/pull/35465/commits/0ead6ef8737421e4323f290ef1d82952dec270a3

Would you mind taking another look when you have a moment? I really appreciate your time and effort in helping get this PR merged

phsantiago commented:
Hi @scottaohara, just following up on this PR as it’s been a few weeks since my last re-review request. I’ve addressed the feedback, and would really appreciate it if you could take another look when you have time. Thanks again!

Josh-Cena commented:
Perhaps @estelle could also review this, if @scottaohara could not
